src/components/WidgetCalendar.vue

```diff
@@ -0,0 +1,156 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+
+const weeks = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+const todayDate = new Date();
+const currentYear = todayDate.getFullYear();
+const currentMonth = todayDate.getMonth();
+
+const formattedMonthYear = computed(() => {
+  const options: Intl.DateTimeFormatOptions = { month: 'short', year: 'numeric' };
+  return todayDate.toLocaleDateString(undefined, options);
+});
+
+const firstDayOfMonth = new Date(currentYear, currentMonth, 1);
+const lastDayOfMonth = new Date(currentYear, currentMonth + 1, 0);
+
+const lastDateOfPrevMonth = new Date(currentYear, currentMonth, 0).getDate();
+
+const weeksInMonth = computed(() => {
+  const weeksArray: Array<Array<{ date: Date; isToday: boolean; isCurrentMonth: boolean }>> = [];
+  let week: Array<{ date: Date; isToday: boolean; isCurrentMonth: boolean }> = [];
+
+  const startDay = firstDayOfMonth.getDay();
+
+  for (let i = startDay; i > 0; i--) {
+    const date = new Date(currentYear, currentMonth - 1, lastDateOfPrevMonth - i + 1);
+    week.push({
+      date,
+      isToday: isSameDate(date, todayDate),
+      isCurrentMonth: false
+    });
+  }
+
+  for (let i = 1; i <= lastDayOfMonth.getDate(); i++) {
+    const date = new Date(currentYear, currentMonth, i)
+    week.push({
+      date,
+      isToday: isSameDate(date, todayDate),
+      isCurrentMonth: true
+    });
+
+    if (week.length === 7) {
+      weeksArray.push(week);
+      week = [];
+    };
+  }
+
+  if (week.length > 0) {
+    for (let i = 1; week.length < 7; i++) {
+      const date = new Date(currentYear, currentMonth + 1, i);
+      week.push({
+        date,
+        isToday: isSameDate(date, todayDate),
+        isCurrentMonth: false
+      });
+    }
+    weeksArray.push(week);
+  }
+
+  return weeksArray;
+})
+
+const isSameDate = (date1: Date, date2: Date): boolean => {
+  return (
+    date1.getFullYear() === date2.getFullYear() &&
+    date1.getMonth() === date2.getMonth() &&
+    date1.getDate() === date2.getDate()
+  );
+}
+</script>
+
+<template>
+  <div class="calendar">
+    <div class="header">
+      <h2>{{ formattedMonthYear }}</h2>
+    </div>
+    <table>
+      <thead>
+        <tr>
+          <th v-for="week in weeks" :key="week">{{ week }}</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr v-for="(week, index) in weeksInMonth" :key="index">
+          <td v-for="day in week" :key="day.date.getTime()" :class="{
+            today: day.isToday,
+            sunday: day.date.getDay() === 0,
+            saturday: day.date.getDay() === 6,
+            'is-current-month': day.isCurrentMonth
```

Comment:
```suggestion
            [$style.today]: day.isToday,
            [$style.sunday]: day.date.getDay() === 0,
            [$style.saturday]: day.date.getDay() === 6,
            [$style.isCurrentMonth]: day.isCurrentMonth
```
こうするとstyle module使えます

Reply:
かぶったわね